head.go

```diff
@@ -587,8 +587,12 @@ func (h *Head) Delete(mint, maxt int64, ms ...labels.Matcher) error {
 	for p.Next() {
 		series := h.series.getByID(p.At())
 
+		t0, t1 := series.minTime(), series.maxTime()
+		if t0 < 0 || t1 < 0 {
```

Comment:
Why do we check `< 0` here but set `math.MinInt64` below. If we want to have the smallest number as a sentinel value checks should be precise for it as well.

That said, while we theoretically support timestamps below `0` practically it won't end well.

Reply:
Done